filebeat/harvester/reader/multiline.go

```diff
@@ -114,31 +114,34 @@ func (mlr *Multiline) Next() (Message, error) {
 
 func (mlr *Multiline) readFirst() (Message, error) {
 	for {
-		p, err := mlr.reader.Next()
-		if err == nil {
-			if p.Bytes == 0 {
+		message, err := mlr.reader.Next()
+		if err != nil {
+			// no lines buffered -> ignore timeout
+			// TODO: Does this only happen on backoff? Normally is EOF?
+			if err == errMultilineTimeout {
 				continue
 			}
 
-			mlr.startNewLine(p)
-			mlr.state = (*Multiline).readNext
-			return mlr.readNext()
+			// something is wrong here
+			return message, err
 		}
 
-		// no lines buffered -> ignore timeout
-		if err == errMultilineTimeout {
+		// Empty starting lines are skipped
+		if message.Bytes == 0 {
```

Comment:
it's no error. multline doesn't know care about underlying reader/layer. But messages with Bytes==0 are not really interesting to multiline reader.

But: support for fields to message has been added much later. So, what shall we do if fields are still set, but Bytes==0? Maybe handling should be changed to store fields of message if Bytes==0. One can even consider passing this message along if buffer is still empty.